### PR TITLE
WIP: changes to make rop search work with Thumb

### PR DESCRIPTION
**Pull request by radare**
cc @arnaugamez

### CODE REVIEW

File: libr/core/cmd_search.c [hunk 1]
```diff
@@ -1039,9 +1041,11 @@ static RList *construct_rop_gadget(RCore *core, ut64 addr, ut8 *buf, int idx, co
 	}
 	int opsz = 0;
 	char *opst = NULL;
+int minopsz = 2;
```
Comment (lowlyw) [inline libr/core/cmd_search.c]: fix indent

File: libr/core/cmd_search.c [hunk 2]
```diff
@@ -1229,7 +1237,10 @@ static void print_rop(RCore *core, RList *hitlist, char mode, bool *json_first)
 			buf[hit->len] = 0;
 			r_io_read_at (core->io, hit->addr, buf, hit->len);
 			r_asm_set_pc (core->assembler, hit->addr);
+			r_asm_set_bits (core->assembler, 16);
```
Comment (lowlyw) [inline libr/core/cmd_search.c]: uh, wat, is this always disassembling as 16 bit, that's not going to fly

File: libr/core/cmd_search.c [hunk 3]
```diff
@@ -1039,9 +1041,11 @@ static RList *construct_rop_gadget(RCore *core, ut64 addr, ut8 *buf, int idx, co
 	}
 	int opsz = 0;
 	char *opst = NULL;
+int minopsz = 2;
 
 	while (nb_instr < max_instr) {
 		ht_uu_insert (localbadstart, idx, 1);
+		r_asm_set_bits (core->assembler, 16);
```
Comment (lowlyw) [inline libr/core/cmd_search.c]: uh, wat, is this always disassembling as 16 bit, that's not going to fly

**Comment by radare** [comment]
This is the poc to show that it fixes my current use case for this. I can share the testcase and can be pushed to r2r without issues. I know this pr is wrong way to solve and it breaks things thats why its WIP.

File: libr/core/cmd_search.c (hunk 1 above)
Comment (radare) [reply libr/core/cmd_search.c]: fixed and improved this hardcoded value

File: libr/core/cmd_search.c (hunk 3 above)
Comment (radare) [reply libr/core/cmd_search.c]: well. i think we had a config var to specify that we dont want cpu mode switches iirc

**Comment by radare** [comment]
ping @arnaugamez

**Comment by ret2libc** [comment]
If this is just WIP, we should just close it.

**Comment by radare** [comment]
there's no issue to track this afaik, i was waiting for about 1 year for arnau to test this thing. i will come back at some point because this needs to be fixed, but i have some extra ideas to improve esil analysis with recursive queues and this is related, so keep it. i take care of my own prs, even if they seem abandoned for years

**Comment by ret2libc** [comment]
It's just after 1year the code change... And keeping a PR opened for 1 year without active work is just confusing IMHO :/ but ok.

**Comment by radare** [comment]
last week i finished a pr that was open from me for more than 2 years. im confusing

**Comment by ret2libc** [comment]
Closing. This had no review, no changes, no nothing for a long time. I'm closing the PR as there is no benefit in keeping this open. The branch will still be active, so if @trufae wants he can continue working on this when he has time. No need to keep it in the backlog.

**Pull request closed without merging**